pkg/grpc/api.go

```diff
@@ -212,16 +214,56 @@ func (a *api) InvokeBinding(ctx context.Context, in *runtimev1pb.InvokeBindingRe
 	return r, nil
 }
 
-func (a *api) GetState(ctx context.Context, in *runtimev1pb.GetStateRequest) (*runtimev1pb.GetStateResponse, error) {
+func (a *api) GetBulkState(ctx context.Context, in *runtimev1pb.GetBulkStateRequest) (*runtimev1pb.GetBulkStateResponse, error) {
+	store, err := a.getStateStore(in.StoreName)
+	if err != nil {
+		return &runtimev1pb.GetBulkStateResponse{}, err
+	}
+
+	var wg sync.WaitGroup
+	resp := &runtimev1pb.GetBulkStateResponse{}
+
+	for _, k := range in.Keys {
+		wg.Add(1)
+
+		go func(k string) {
```

Comment:
We may have to limit the concurrency here. Otherwise, Some of state store would encounter throttling issue.

Reply:
How do you think concurrency limiting here should work?

Reply:
We can optionally add a `Parallelism` field to the request object.

Reply:
yup adding option makes sense to me.

Reply:
But we may need to limit 5 or 10 get operation by default.

Reply:
Great, I'll add the parallelism field.

By default we should probably not limit, that's the default behavior users get in most programming language, C# Parallel.For as an example.

Reply:
Added.